### DISCUSSION TITLE
Add button to toggle tab menu

### DISCUSSION THREAD

**Discussion by tobias-klein** [ezra-bible-app/ezra-bible-app#452] [Ideas]
See discussion #349.

On smaller screens and also in general for more focus it would be nice to have a way to toggle the tab menu. This could be done with a small button directly underneith the full screen button.

**Comment by augustin-colesnic** [comment]
The fullscreen button itself should do that @tobias-klein. Tab menu should be removed on fullscreen mode. I would love to have that in the next release :)

**Comment by tobias-klein** [comment]
I can see your point, Augustin. However, the question is what functionality then remains in full screen. What you describe would be a read-only mode. Unless you provide the possibility of assigning tags via verse-based context menus. Even when going fullscreen I personally would still like to be able to assign tags.

**Comment by augustin-colesnic** [comment]
Yes, tags could be assigned only in full screen if we implement a better visually UX/UI. I am thinking of a circle/ wheel of options when a verse is long pressed.  It may be something like in this gif:   https://images.app.goo.gl/RG4b9GabiovN92SP7

**Comment by tobias-klein** [comment]
Having to long press or right click on the desktop is only a last resort in my mind ... if somehow possible we should make most of the functionality available via regular left click / tap. Otherwise this will be rather hidden to first time users.

@augustin-colesnic Do you know a good and well maintained vanilla JS library that creates context menus?

**Comment by augustin-colesnic** [comment]
@tobias-klein  try these: https://www.cssscript.com/tag/circle-menu/

**Reply by tobias-klein** [reply]
Since this is still a bit fuzzy I converted the issue into a discussion.

I just played a bit with this one:
https://github.com/axln/radial-menu-js

This is what I came up with (really quickly ...)

![image](https://user-images.githubusercontent.com/17739895/134366455-336d5ca2-287a-4875-92dc-e98c4913f4a4.png)

I think it would be nice to still have some text for each menu item (not only icons), like in this example.
This little library uses SVG, which makes it look really nice and also scale nicely. Also, if additional elements are needed the code is very easy to adjust (let's say six instead of four slices).

Some of those other examples I found are based on static pixel-based offsets and I think that's a no-go ...

What's not covered by this library is the transition from a button to this menu, but that could be easily developed.

@augustin-colesnic What do you think?

**Reply by tobias-klein** [reply]
Other potential candidates:
https://github.com/victorqribeiro/radialMenu (only supports text)
http://wheelnavjs.softwaretailoring.net/index.html (Quite sophisticated, maybe a bit too much ...)

**Reply by augustin-colesnic** [reply]
great! i would go with axln radial menu you already gave a run. gg👌🏻

**Reply by tobias-klein** [reply]
This is a demo in the context of Ezra.

![image](https://user-images.githubusercontent.com/17739895/134578995-6c08246a-7590-4463-883b-3c2113990a3f.png)

You would click on the verse reference and then the menu would appear.
Feedback?

**Comment by augustin-colesnic** [comment]
perfect! Go for it buddy! :)

**Comment by tobias-klein** [comment]
Further discussion of technical details should happen in the context of the [ongoing pull request](https://github.com/ezra-bible-app/ezra-bible-app/pull/454).
